### PR TITLE
fix: helm/docker/maven path error

### PR DESCRIPTION
**Pull request by yyzxw**
修复helm,docker和maven因为路径不对无法直接使用的问题

### CODE REVIEW

File: jenkins-agent/base/ansible/playbooks/tasks/common.yaml [hunk 1]
```diff
@@ -9,94 +13,100 @@
...
   retries: 3
   delay: 5
 
 - name: Install Node.js
   shell: |
-    curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.40.0/install.sh | bash
-    . "$NVM_DIR/nvm.sh"
-    nvm install {{ node_version }}
-    nvm alias default {{ node_version }}
+    curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.40.0/install.sh | bash
```
Comment (muma378) [inline jenkins-agent/base/ansible/playbooks/tasks/common.yaml:102]: imo nvm is not the best practice to install nodejs in container. 

It is more like `virtualenv` in python, a development environment management tool to allow multiple versions coexist in a host. But in container you have no such requirement, you just pull another image ...

File: jenkins-agent/maven/Dockerfile [hunk 2]
```diff
@@ -16,7 +18,7 @@ RUN ansible-playbook ./ansible/playbooks/main.yaml -vvv -e java_version=$JAVA_VE
 # maven
 ENV MAVEN_VERSION 3.9.9
```
Comment (muma378) [inline jenkins-agent/maven/Dockerfile]: too many warnings, use `ENV MAVEN_VERSION=3.9.9`

File: jenkins-agent/maven/Dockerfile [hunk 3]
```diff
@@ -2,11 +2,13 @@ ARG RUNTIME
 ARG REGISTRY_REPO
```
Comment (muma378) [inline jenkins-agent/maven/Dockerfile]: maybe set default value? ref: https://docs.docker.com/reference/build-checks/invalid-default-arg-in-from/

File: jenkins-agent/base/ansible/playbooks/tasks/common.yaml [hunk 4]
```diff
@@ -1,6 +1,10 @@
 - name: Install Docker
```
Comment (muma378) [inline jenkins-agent/base/ansible/playbooks/tasks/common.yaml:6]: maybe remove the pre-installed docker before installing?

File: jenkins-agent/base/ansible/playbooks/tasks/centos/python.yaml [hunk 5]
```diff
@@ -17,27 +23,32 @@
 
 - name: Install Python from source
   shell: |
-    wget https://www.python.org/ftp/python/{{ python_version }}/Python-{{ python_version }}.tgz && \
-    tar xvf Python-{{ python_version }}.tgz && \
-    cd Python-{{ python_version }} && \
+    wget -q https://www.python.org/ftp/python/{{ python_version }}/Python-{{ python_version }}.tgz && \
```
Comment (muma378) [inline jenkins-agent/base/ansible/playbooks/tasks/centos/python.yaml:26]: do we have to compile from source·····

File: jenkins-agent/base/ansible/playbooks/tasks/common.yaml [hunk 6]
```diff
@@ -9,94 +18,100 @@
...
-    curl -Lo /tmp/sonar_scanner.zip https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-{{ sonar_scanner_version }}.zip &&
-    unzip /tmp/sonar_scanner.zip -d /opt/ &&
-    mv /opt/sonar-scanner-{{ sonar_scanner_version }} /opt/sonar-scanner &&
-    sed -i '1 a JAVA_HOME=/opt/java-{{ jdk_11_version_dir }}' /opt/sonar-scanner/bin/sonar-scanner &&
-    echo 'export PATH=/opt/sonar-scanner/bin:$PATH' >> /etc/profile.d/sonar-scanner.sh &&
+    curl -Lo /tmp/sonar_scanner.zip https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-{{ sonar_scanner_version }}.zip &&  
+    unzip /tmp/sonar_scanner.zip -d /opt/ &&  
+    mv /opt/sonar-scanner-{{ sonar_scanner_version }} /opt/sonar-scanner &&  
+    sed -i '1 a JAVA_HOME=/opt/java-{{ jdk_11_version_dir }}' /opt/sonar-scanner/bin/sonar-scanner &&  
+    echo 'export PATH=/opt/sonar-scanner/bin:$PATH' >> /etc/profile.d/sonar-scanner.sh &&
```
Comment (muma378) [inline jenkins-agent/base/ansible/playbooks/tasks/common.yaml:83]: how /etc/profile.d/sonar-scanner.sh get work

File: jenkins-agent/base/ansible/playbooks/tasks/common.yaml [hunk 7]
```diff
@@ -9,94 +18,100 @@
...
 - name: Install Go
   shell: |
-    wget https://golang.google.cn/dl/go{{ go_version }}.linux-{{ arch_amd64 or arch_arm64 }}.tar.gz
-    tar -xvf go{{ go_version }}.linux-{{ arch_amd64 or arch_arm64 }}.tar.gz
-    rm -rf go{{ go_version }}.linux-{{ arch_amd64 or arch_arm64 }}.tar.gz
-    mv go /usr/local/go
+    wget https://golang.google.cn/dl/go{{ go_version }}.linux-{{ arch_amd64 or arch_arm64 }}.tar.gz  
+    tar -xvf go{{ go_version }}.linux-{{ arch_amd64 or arch_arm64 }}.tar.gz  
+    rm -rf go{{ go_version }}.linux-{{ arch_amd64 or arch_arm64 }}.tar.gz  
+    mv go /usr/local/go
```
Comment (muma378) [inline jenkins-agent/base/ansible/playbooks/tasks/common.yaml:94]: do we need to add /usr/local/go/bin to PATH?

**Review by muma378 — APPROVED** [review]

File: jenkins-agent/base/ansible/playbooks/tasks/common.yaml (hunk 7 above)
Comment (yyzxw) [reply jenkins-agent/base/ansible/playbooks/tasks/common.yaml:94]: 在env中有的
image

File: jenkins-agent/base/ansible/playbooks/tasks/common.yaml (hunk 6 above)
Comment (yyzxw) [reply jenkins-agent/base/ansible/playbooks/tasks/common.yaml:83]: 以前安装的时候是这么装的，不过看这篇文章https://blog.csdn.net/m0_37649480/article/details/141054810 其实就是配了一下sonar-scaner的路径

**Pull request merged**